quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocomplete-adapter-utills.ts

```diff
@@ -0,0 +1,164 @@
+  const allChildren = [];
+
+  for (const extractedTable of tables) {
+    const { columns } = extractedTable;
+
+    for (const column of columns) {
+      if (typeof column === 'object') {
+        if (typeof column.dataType === 'string') {
+          column.dataType = trinoToJs(column.dataType, 0);
+        }
+        if (typeof column.dataType === 'object') {
+          allChildren.push(...getObjectChildren(column.dataType, column.name));
+        }
+      }
+    }
+  }
+
+  return allChildren;
+}
+
+
+export function getSearchCompletion(tables: TableInfo[], prefix: string | undefined): any {
+  if (!prefix.includes('.')) {
+    return [];
+  }
+  const allChildren = getAllChildrenOfTables(tables);
+  const relevantPartOfPrefix = findRelevantPartOfPrefix(tables, prefix.split('.')).slice(0, -1);
+  if (!relevantPartOfPrefix) {
+    return []
+  }
+  const lastDotIndex = findLastDotIndex(relevantPartOfPrefix);
+  const startOfSearch = lastDotIndex !== -1 ? relevantPartOfPrefix.slice(0, lastDotIndex + 1) : relevantPartOfPrefix;
+  const searchPart = relevantPartOfPrefix.replace(startOfSearch, '')
+  const filteredChildren = allChildren.filter(obj => {
+    const lowerCasedName = obj.name.toLowerCase();
+    const parts = lowerCasedName.split('.');
+    if (parts.length > 1) {
+      const substringAfterFirstDot = parts.slice(1).join('.');
+      const criteria = checkCriteria(substringAfterFirstDot, searchPart.toLowerCase());
+      const filterIfInDifferentColumn = obj.name.startsWith(relevantPartOfPrefix.split('.')[0]);
+      return obj.name.includes(startOfSearch) && substringAfterFirstDot.includes(searchPart.toLowerCase()) && criteria && filterIfInDifferentColumn;
+    }
+    return false;
+  });
+  const prefixUntilLastDot = extractPrefixUntilLastDot(relevantPartOfPrefix);
+  const completionArray = filteredChildren.map(obj => ({
+    value: obj.name,
+    meta: typeof obj.dataType === 'object' ? 'row' : obj.dataType,
+    caption: obj.name.slice(prefixUntilLastDot.length + 1)
+  }));
+  return completionArray;
+}
+
+export function getNextLevel(tables: TableInfo[], prefix: string | undefined): any {
+  const allChildren = getAllChildrenOfTables(tables);
+  const relevantPartOfPrefix = findRelevantPartOfPrefix(tables, prefix.split('.'));
+  const relevantChildren = allChildren.filter(obj => {
+    const dotCount = obj.name.split('.').length - 1;
+    return obj.name.startsWith(relevantPartOfPrefix) && dotCount === relevantPartOfPrefix.split('.').length - 1;
+  });
+  const completionArray = relevantChildren.map(obj => ({
+    value: prefix.replace(relevantPartOfPrefix, '') + obj.name,
+    meta: typeof obj.dataType === 'object' ? 'row' : obj.dataType,
+    caption: obj.name.slice(relevantPartOfPrefix.length)
+  }));
+  return completionArray
+}
+
+function checkCriteria(substringAfterFirstDot, searchPart) {
+  const lastDotIndex = findLastDotIndex(substringAfterFirstDot);
+
+  if (lastDotIndex !== -1) {
```

Comment:
try to simplify
and replace to lastDotIndex >= 0

Reply:
Simplified  too:
```function checkCriteria(substringAfterFirstDot, searchPart) {
  const lastDotIndex = findLastDotIndex(substringAfterFirstDot);
  return lastDotIndex >= 0 ? substringAfterFirstDot.substring(lastDotIndex + 1).includes(searchPart) : substringAfterFirstDot.includes(searchPart);
}